pkg/controller/model/grafanaServiceAccount.go

```diff
@@ -23,6 +23,24 @@ func getServiceAccountAnnotations(cr *v1alpha1.Grafana, existing map[string]stri
 	return MergeAnnotations(cr.Spec.ServiceAccount.Annotations, existing)
 }
 
+func mergeImagePullSecrets(requested []v1.LocalObjectReference, existing []v1.LocalObjectReference) []v1.LocalObjectReference {
+	if existing == nil {
+		return requested
+	}
+
+	for k, v := range requested {
```

Comment:
@AdheipSingh  @HubertStefanski  Sorry, one last question: this code seems to be taken from [here](https://github.com/integr8ly/grafana-operator/blob/master/pkg/controller/model/utils.go#L22) where it's merging two maps where the index is the name of the annotation.

Here however, `existing` and `requested` are two arrays, so the index is the position within the array. This means every entry in `existing` is overwritten with the value in the same position in `requested`. And it can lead to prepopulated (by openshift) elements being overwritten and causing the same conflict.

I think a better strategy would be to merge the `LocalObjectReferences`s by their names: make sure the resulting array contains all elements of both arrays without duplicates.

What do you think?

Reply:
@HubertStefanski @AdheipSingh what do you think about this instead:

```
func mergeImagePullSecrets(requested []v1.LocalObjectReference, existing []v1.LocalObjectReference) []v1.LocalObjectReference {
	appendIfAbsent := func(secrets []v1.LocalObjectReference, secret v1.LocalObjectReference) []v1.LocalObjectReference {
		for _, s := range secrets {
			if s.Name == secret.Name {
				return secrets
			}
		}
		return append(secrets, secret)
	}

	for _, s := range requested {
		existing = appendIfAbsent(existing, s)
	}

	return existing
}
```

Reply:
@pb82 Sure, that seems more robust, I'll spin up an OSD cluster first thing in the morning and try this out, @AdheipSingh if you'd like to make the changes? or if we're short on time I can do it, I don't mind either way. Once this is pushed and verified we can finally put together that release.

Reply:
@HubertStefanski please go ahead and do the changes, i am not an openshift user. @pb82 the func seems fine, it will be better to deal with localobjectreference as input params. 
The strange part is somehow my changes are working fine in my cluster :|

Reply:
@AdheipSingh I think this is because OpenShift automatically adds its own image pull secrets to the SA and that does not happen in Minikube or other environments.

Reply:
i see, makes sense